[error]Cannot bind argument to parameter 'Path' because it is null at selfhosted agent

```
2020-02-16T12:58:39.2328778Z ##[section]Starting: Promote to Prerelease
2020-02-16T12:58:39.2448778Z ==============================================================================
2020-02-16T12:58:39.2448778Z Task         : Promote package to Release View
2020-02-16T12:58:39.2448778Z Description  : Promotes a package to a Release View in VSTS Package Management
2020-02-16T12:58:39.2448778Z Version      : 3.0.2
2020-02-16T12:58:39.2448778Z Author       : Rene van Osnabrugge
2020-02-16T12:58:39.2448778Z Help         : Version: 3.0.2. [More Information](https://github.com/renevanosnabrugge/vsts-promotepackage-task/wiki)
2020-02-16T12:58:39.2448778Z ==============================================================================
2020-02-16T12:58:40.5108778Z Promoting package(s) by reading metadata from package file(s) matching pattern '**/*.nupkg !**/*.symbols.nupkg' from root directory 'C:\agent\_work\28\a'
2020-02-16T12:58:40.9158778Z Matching paths found:
2020-02-16T12:58:40.9158778Z C:\agent\_work\28\a\NeosyntezApiClient.1.0.0-CI-20200216-125819.nupkg C:\agent\_work\28\a\NeosyntezApiClient.AutofacModule.1.0.0-CI-20200216-125819.nupkg C:\agent\_work\28\a\NeosyntezApiClient.Domain.1.0.0-CI-20200216-125819.nupkg
2020-02-16T12:58:41.0938778Z ##[error]Cannot bind argument to parameter 'Path' because it is null.
2020-02-16T12:58:41.1358778Z ##[section]Finishing: Promote to Prerelease
```

The yml task is 

```
- task: rvo-vsts-promotepackage-task@3
  displayName: Promote to Prerelease
  inputs:
    feed: 'Martinov_NugetService'
    inputType: 'packageFiles'
    packagesDirectory: '$(Build.ArtifactStagingDirectory)'
    packagesPattern: |
      **/*.nupkg
      !**/*.symbols.nupkg
    releaseView: 'e35d64ae-f9db-4fc4-a9ad-b1b71b572826'
```

How can I avoid the error?

--- a/vsts-promotepackage-task/vsts-promotepackage-task.ps1
+++ b/vsts-promotepackage-task/vsts-promotepackage-task.ps1
@@ -216,12 +216,34 @@ function New-PackageObject {
     return $package
 }
 
+function Test-CommandExists
+{
+    param ($command)
+    $oldPreference = $ErrorActionPreference
+    $ErrorActionPreference = ‘stop’
+
+    try {
+        if ( Get-Command $command ){ return $true }
+    }
+    catch {}
+    finally { 
+        $ErrorActionPreference=$oldPreference 
+    }
+
+    return $false
+}
+
 Add-Type -AssemblyName System.IO.Compression.FileSystem
 function Get-NuGetPackageMetadata([string]$filePath) {
     $package = New-PackageObject
     $zip = [System.IO.Compression.ZipFile]::OpenRead($filePath)
     try {
-        $tempFilePath = (New-TemporaryFile).FullName
+        # New-TemporaryFile doesn't exist in PS4
+        if (Test-CommandExists "New-TemporaryFile") {
+            $tempFilePath = (New-TemporaryFile).FullName
+        } else {
+            $tempFilePath = [System.IO.Path]::GetTempFileName()
+        }
         $nuspecEntry = $zip.Entries | Where-Object { $_.FullName -like '*.nuspec' } | Select-Object -First 1
         [System.IO.Compression.ZipFileExtensions]::ExtractToFile($nuspecEntry, $tempFilePath, $true)
         $xml = [xml](Get-Content $tempFilePath)

